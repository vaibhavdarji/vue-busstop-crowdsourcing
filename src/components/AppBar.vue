<template>
  <div>
    <v-toolbar dark color="primary">
      <v-toolbar-title class="white--text" :id="headerInfo.ELEM_ID">
        {{headerInfo.TITLE}}
      </v-toolbar-title>
      <v-spacer></v-spacer>
      <v-toolbar-items>
        <h1 class="headline py-3" :id="headerInfo.TF_ELEM_ID">
          Total Donation: 
          <v-icon>attach_money</v-icon>
          {{totalFunds}}
        </h1>
      </v-toolbar-items>
    </v-toolbar>
  </div>
</template>

<script>
import { mapGetters } from 'vuex';
import Constants from '@/constants';
export default {
  name: 'AppBar',
  data () {
    return {
      appTitle: 'Bus-stop crowdsourcing web application'
    };
  },

  computed: {
    ...mapGetters([
      'totalFunds'
    ]),

    headerInfo () {
      return Constants.APP_TITLE;
    }
  }
}
</script>

