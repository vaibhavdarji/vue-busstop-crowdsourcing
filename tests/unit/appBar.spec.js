import { shallowMount, mount } from "@vue/test-utils";
import AppBar from "@/components/AppBar.vue";
import Constants from '@/constants';

import Vue from "vue";
import Vuex from "vuex";
import VueRouter from 'vue-router';
import Vuetify from "vuetify";


Vue.use(VueRouter);
Vue.use(Vuetify);
Vue.use(Vuex);

describe("AppBar.vue", () => {
  let getters, store;

  beforeEach(() => {
    getters = {
      totalFunds: () => 0
    };
    store = new Vuex.Store({
      getters
    });
  });
  it("should match snapshot", () => {
    const wrapper = shallowMount(AppBar, {
      store,
      Vue
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should contain app title, render correct title element", () => {
    const wrapper = shallowMount(AppBar, {
      store,
      Vue
    });

    const {
      TITLE,
      ELEM_ID
    } = Constants.APP_TITLE;

    const appTitleElement = wrapper.find(`#${ELEM_ID}`);

    expect(appTitleElement.text()).toBe(TITLE);
  });

  it("should contain total fund, render correct total fund element", () => {
    const wrapper = shallowMount(AppBar, {
      store,
      Vue
    });

    const {
      TF_ELEM_ID
    } = Constants.APP_TITLE;
    const tfElement = wrapper.find(`#${TF_ELEM_ID}`);
    expect(tfElement.text()).toContain(getters.totalFunds().toString());
  });
});
