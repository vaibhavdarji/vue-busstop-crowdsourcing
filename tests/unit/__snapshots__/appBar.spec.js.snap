// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar.vue should match snapshot 1`] = `
<div>
  <v-toolbar-stub
    color="primary"
    dark="true"
    scrollthreshold="300"
  >
    <v-toolbar-title-stub
      class="white--text"
      id="bsc-app-title"
    >
      
      Bus-stop crowdsourcing web application
    
    </v-toolbar-title-stub>
     
    <v-spacer-stub />
     
    <v-toolbar-items-stub>
      <h1
        class="headline py-3"
        id="bsc-app-total-fund"
      >
        
        Total Donation: 
        
        <v-icon-stub>
          attach_money
        </v-icon-stub>
        
        0
      
      </h1>
    </v-toolbar-items-stub>
  </v-toolbar-stub>
</div>
`;
